libs/sdk-core/src/breez_services.rs

```diff
@@ -1924,36 +1924,58 @@ impl BreezServices {
 
     /// Registers for lightning payment notifications. When a payment is intercepted by the LSP
     /// to this node, a callback will be triggered to the `webhook_url`.
+    ///
+    /// Note: these notifications are registered for all LSPs (active and historical) with whom
+    /// we have a channel.
```

Comment:
this function is called by the `connect_lsp` function. I don't think it's necessary to register payment notifications for old LSPs when called from `connect_lsp`

Reply:
On first `connect_lsp`, there are no historical LSPs, so the loop for historical LSPs won't find anything. On subsequent `connect_lsp` calls when changing LSP, I think it makes sense to register notifications for historical LSPs with open channels (that's the main goal of the PR).

So overall I think the behavior is as intended.